coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/sql/SqlSecondaryManifest.scala

```diff
@@ -208,6 +210,49 @@ class SqlSecondaryManifest(conn: Connection) extends SecondaryManifest {
   }
 
   def releaseClaimedDataset(job: SecondaryRecord): Unit = {
+    val savepoint = conn.setSavepoint()
+
+    // retrying with rollback to savepoint
+    def retrying(backoffMillis: Int = 5): Unit = {
+      if (backoffMillis > 300000) { // > 5 minutes
+        log.error("Ran out of retries; failed to release claim on dataset {} in secondary {}!",
+          job.datasetId, job.storeId)
+        throw new Exception("Ran out of retries; Failed to release claim on dataset!")
+      }
+
+      try {
+        log.trace("Attempting to release claim on dataset for job: {}.", job)
+        releaseClaimedDatasetUnsafe(job)
+      } catch {
+        case e: SQLException =>
+          log.warn("Unexpected sql exception while releasing claim on dataset {} in secondary {}",
+            job.datasetId.asInstanceOf[AnyRef], job.storeId, e)
+          conn.rollback(savepoint)
```

Comment:
Just want to confirm that you can rollback to the same savepoint multiple times.